# rs_user_clean

In [1]:
%matplotlib inline
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import io
import os

# rs_user

### usuarios navegando

In [2]:
data_user="../DATA/IntropiaCSV/rs_user.csv"

readme=os.path.join(data_user)
#cat $readme | head

In [3]:
datafile = os.path.join(data_user)
data_user = pd.read_csv(datafile, sep=',')
data_user.head(10)


iduser  tags  geolocation  description  name  active  imageurl  url  \
0  1000004887844   NaN          NaN          NaN   NaN       1       NaN  NaN   
1  2000000000333   NaN          NaN          NaN   NaN       1       NaN  NaN   
2  2000000000334   NaN          NaN          NaN   NaN       0       NaN  NaN   
3  2000000000335   NaN          NaN          NaN   NaN       1       NaN  NaN   
4  2000000000336   NaN          NaN          NaN   NaN       1       NaN  NaN   
5  2000000000337   NaN          NaN          NaN   NaN       1       NaN  NaN   
6  2000000000338   NaN          NaN          NaN   NaN       1       NaN  NaN   
7  2000000000339   NaN          NaN          NaN   NaN       1       NaN  NaN   
8  2000000000340   NaN          NaN          NaN   NaN       0       NaN  NaN   
9  2000000000341   NaN          NaN          NaN   NaN       1       NaN  NaN   

    misc   timestamp  
0     es  1459288467  
1  es_es  1376468347  
2  es_es  1376464186  
3     es  1485442621  
4  gb_en  1376484799  
5     es  1470349078  
6     de  1484577027  
7  gr_en  1376467169  
8  be_en  1378466593  
9     es  1490222814

In [4]:
data_user.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48969 entries, 0 to 48968
Data columns (total 10 columns):
iduser         48969 non-null int64
tags           0 non-null float64
geolocation    0 non-null float64
description    0 non-null float64
name           0 non-null float64
active         48969 non-null int64
imageurl       0 non-null float64
url            0 non-null float64
misc           48968 non-null object
timestamp      48969 non-null int64
dtypes: float64(6), int64(3), object(1)
memory usage: 3.7+ MB


In [5]:
# Hay muchas coluans sin valores. No nos aportan nada.

In [6]:
# seleccionamos columnas interés.

user = data_user[['iduser','active','misc','timestamp']]

# vistazo

user.head()

iduser  active   misc   timestamp
0  1000004887844       1     es  1459288467
1  2000000000333       1  es_es  1376468347
2  2000000000334       0  es_es  1376464186
3  2000000000335       1     es  1485442621
4  2000000000336       1  gb_en  1376484799

In [7]:
# Calculate number of diferents elements >> unicos

n_user =data_user.iduser.unique().shape[0]
n_active = data_user.active.unique().shape[0]
n_misc = data_user.misc.unique().shape[0]

print("There are %s users" % (n_user))
print("There are %s active" % (n_active))
print("There are %s misc" % (n_misc))

There are 48969 users
There are 2 active
There are 171 misc


Notas: Tenemos menos emails que usuarios, esto puede deberse a que son guest o bien tienen cuenta.
En cualquier caso, no nos importa, ya que lo relevante es el "id".
Tenemos una combinación de 171 misc. Esta es una combinación de país-idioma por site. 
Hay web site que tienen 2 idiomas y otros que tienen 1.
Puede ser relevante a la hora de recomendar, dados loa hábitos de compra, climatología...pero inicialemente
no lo contemplo.
    


In [8]:
# otra forma de hacerlo, mediante una función q retorne 

def count_unique(column):
    return len(column.unique())

data_user[['iduser','active','misc']].apply(count_unique, axis=0).astype(np.int32)

iduser    48969
active        2
misc        171
dtype: int32

In [9]:
# Vistazo a la cantidad de emails que son recurrentes.

# email_count = user.groupby('email').size().sort_values(ascending=False)
# print (email_count)

# Esta columna tiene relevancia solo para uso interno:
# Reflexión para dto. de marketing:
# Tenemos muy pocos emails recurrentes, esto se puede traducir en q tenemos muy pocos usuarios recurrentes???
# La bbbdd de emails es correcta >> revisar de forma conjunta

In [10]:
# Vistazo rápido activo/no activo.>> deduzco q es un indicador de un estado interno.

active_count = user.groupby('active').size().sort_values(ascending=False)
print (active_count)

active
1    41873
0     7096
dtype: int64


Sobre la columna "misc" >> que significan las útimas URL's??? se trata de enlaces rotos:

Posibilidades:

- vienen de emails recuperación carritos? 
- se deben a la incidencia que hubo en la página respecto al redireccionamiento? >> comprobar.

En cualquier caso, inicialmente no se tiene en cuenta para ningún calculo importante.

In [11]:
misc = user['misc'].value_counts()
misc


es                                                                                                     33048
es_es                                                                                                   5086
gb                                                                                                      2584
pt                                                                                                      1286
it                                                                                                      1075
gb_en                                                                                                    859
fr                                                                                                       795
de                                                                                                       787
it_en                                                                                                    405
pt_en              

In [12]:
# Explorando

In [13]:
result = user['active'] == 0
len(result)

48969

In [14]:
len(user[result])

7096

In [15]:
user[result].head()

iduser  active   misc   timestamp
2   2000000000334       0  es_es  1376464186
8   2000000000340       0  be_en  1378466593
10  2000000000342       0  pt_en  1421016563
11  2000000000343       0  it_en  1376468432
14  2000000000346       0  es_es  1412971285

In [16]:
# Cositas varias probando.

In [17]:
len(user)
user.shape

(48969, 4)

In [157]:
user.count()

iduser       48969
active       48969
misc         48968
timestamp    48969
dtype: int64